### DISCUSSION TITLE
Max. Leistung aller WRs zusammen anpassen

### DISCUSSION THREAD

**Discussion by Troubadix** [tbnobody/OpenDTU#1608] [Ideas]
Hallo,

ich nutze einen HM-800 und einen HM-300. Die daran angeschlossenen Module sind unterschiedlich ausgerichtet, somit kann ich zu verschiedenen Tageszeiten Strom produzieren.

Um die Mittagszeit produzieren aber für einige Zeit beide WRs ordentlich Strom und die eingespeiste Leistung überschreitet die 800 Watt.

In OpenDTU kann ich für jeden WR eine max. Leistung einstellen. 
Ich wünsche mir eine Option, um eine max. Leistung für alle WRs zusammen (Gesamtleistung) einstellen zu können.
Also wenn WR1 700W erzeugt, darf WR2 nur 100W erzeugen. Erzeugt WR1 >500W, darf WR2 100% erzeugen.
Das Limit des jeweiligen WRs soll also von OpenDTU dynamisch angepasst werden, je nach dem was der andere WR gerade macht.

...oder gibt es die Funktion schon und ich hab das übersehen?

**Comment by D3R-ST3FAN** [comment]
Diese "Intelligenz" kannst du doch extern mit jeder Home-Automation Software abbilden. 
PS: auch mit dieser Steuerung wird dein Setup nicht "legal", aber werden denn beide WR realistisch jemals ihre maximale Leistung erreichen?

**Reply by stefan123t** [reply]
@Troubadix please double check the **Dynamic Power Limiter (DPL)** function in the daughter project[ OpenDTU-OnBattery](https://github.com/helgeerbe/OpenDTU-OnBattery/).

I have described a [possible PM+DPL setup](https://github.com/tbnobody/OpenDTU/issues/272#issuecomment-2219644352) with a Shelly Pro3EM as **Power Meter (PM)** here in issue #272.

**Comment by stefan123t** [comment]
@Troubadix schau Dich mal bei OpenDTU-onBattery um, dort gibt es ein DPL (Dynamic Power Limit).
Ob das die von Dir erwünschte Funktion bietet weiß ich natürlich nicht.

https://github.com/helgeerbe/OpenDTU-OnBattery/

Ich habe nochmal drüber nachgedacht, eigentlich wäre es schon praktisch, wenn eine derartige Funktion bereits in OpenDTU intergriert wäre:
* Man kann mehrere Inverter einem PV-Pool bzw. PV-PowerPlant zuordnen.
* Man kann dann (u.a. zeitbasiert z.B. von 18:00 - 7:00 Uhr bzw. basierend auf der Day/Night Time Regelung) ein gemeinsames Limit für einen dieser PV-Pool(s) vorgeben.

* Die OpenDTU vergleicht in jedem Abfragezyklus die aktuellen Werte von den Wechselrichtern mit dem aktuellen PV-Pool Limit.
* Sollte das Limit des PV-Pools mit den aktuellen Werten nicht erreicht werden 
  **und** der aktuell abgefragte Wechselrichter ist mit der Produktion einer seiner Kanäle bereits am oberen Limit, 
  dann wird das Limit dieses WR um 1-5% oder sogar 10% (also ~ 5-50W Schritte) nach oben gesetzt.
* Sollte das Limit des PV-Pools ausgereizt / überschritten sein 
  **und** der aktuell abgefragte Wechselrichter ist mit der Leistung eines seiner Kanäle (knapp) unterhalb des eigenen Limits, 
  so wird das Limit dieses WR um 1-5% oder sogar 10% (also ~ 5-50W Schritte) nach unten gesetzt.

Vorteil dieses Konzeptes ist dass sich das Power Limit dynamisch in 5-50W Schritten in der Nähe (bzw. unterhalb) des Limits für diesen PV-Pool einpendelt. 
Sollte also die Sonneneinstrahlung über den Tag - auf den an einem PV-Pool beteiligten Wechselrichtern und deren Kanälen - das Limit erreichen wird abgeregelt, 
ansonsten werden die Wechselrichter, die bereits ihr Limit auf einem Kanal erreicht haben, etwas weniger limitiert, es wird nachgesteuert.

**Reply by stefan123t** [reply]
Hier eine Übersicht der bei AhoyDTU und OpenDTU verfügbaren MQTT Topics um sowohl das Limit pro Inverter zu setzen als auch die aktuellen Werte der Inverter bzw. MPPT Channels der Inverter abzufragen. Wie man sehen kann sind die meisten Topics sowohl auf AhoyDTU als auch OpenDTU Seite vorhanden, nur eben anders benannt.

Was es bis jetzt weder bei OpenDTU noch bei AhoyDTU gibt ist das o.a. Konzept eines PV-Pools in dem man mehrere Inverter zusammenfassen kann. Es gibt bisher lediglich die Option einzelne Inverter von der Globalen Totalwert auszuschließen.

MQTT-Analyse zur Angleichung Topics von AhoyDTU/OpenDTU
https://github.com/lumapu/ahoy/issues/1212

**Comment by Troubadix** [comment]
> Diese "Intelligenz" kannst du doch extern mit jeder Home-Automation Software abbilden.

Wie? Gibt es dazu ein Howto/Doku, welche Befehle man auf welche Art dazu an OpenDTU schicken muss?

> PS: auch mit dieser Steuerung wird dein Setup nicht "legal", erreichen?

Warum sollte das nicht legal sein? Ich darf (wenn die Gesetzesänderungen endlich mal durch sind) 800W einspeisen (noch 600). Mit wie vielen Modulen und WRs diese Leistung erzeugt wird, ist doch schnuppe. Und wenns mehr sind, gehts halt nicht mehr an der Steckdose und muss gewerblich gemacht werden. Aber auch das ist dann legal.

> aber werden denn beide WR realistisch jemals ihre maximale Leistung 

Die max. Leistung beider WR zusammen werde ich nie erreichen, zusammen werde ich aber über 800W kommen wenn die Sonne ordentlich knallt.

> Schau Dich mal bei OpenDTU-onBattery um, dort gibt es ein DPL (Dynamic Power Limit).
Danke, das sieht interessant aus. Ohne zusätzliche Hardware wäre es aber noch cooler.

Du hast das gut ausgeführt mit Deinen Punkten, kann da nur zustimmen. Genau so hatte ich mir das vorgestellt. Danke!

**Comment by vandinter** [comment]
+1 
Wäre auch sehr an einer Lösung für das beschriebene Szenario, idealerweise direkt über OpenDTU, interessiert.

Danke Euch und Gruß

**Comment by D3R-ST3FAN** [comment]
Zum Thema Legal, damit meinte ich die vereinfachte Anmeldung bzw. was als Steckersolargerät zählt.
Dass du "normal" anmelden kannst ist klar. Dann wird es halt etwas aufwändiger.

Hier  ist der Gesetzentwurf [Solarpaket I](https://dserver.bundestag.de/btd/20/086/2008657.pdf)

>Warum sollte das nicht legal sein? Ich darf (wenn die Gesetzesänderungen endlich mal durch sind) 800W einspeisen (noch 600). Mit wie vielen Modulen und WRs diese Leistung erzeugt wird, ist doch schnuppe. Und wenns mehr sind, gehts halt nicht mehr an der Steckdose und muss gewerblich gemacht werden. Aber auch das ist dann legal.

Da gibt es sehr wohl ein Limit von 2 Kilowatt für die Modulleistung.
Und vom WR 800VA. Also wären 800VA + 300VA davon NICHT abgedeckt.

„(5a) Ein Steckersolargerät oder mehrere Steckersolargeräte mit einer installierten Leistung von
insgesamt bis zu 2 Kilowatt und einer Wechselrichterleistung von insgesamt bis zu 800 Voltampere,
die hinter der Entnahmestelle eines Letztverbrauchers betrieben werden und der unentgeltlichen Abnahme zugeordnet werden, können unter Einhaltung der für die Ausführung eines Netzanschlusses maßgeblichen Regelungen angeschlossen werden. Registrierungspflichten nach der Marktstammdatenregisterverordnung bleiben unberührt; zusätzliche gegenüber dem Netzbetreiber abzugebende Meldungen
von Anlagen nach Satz 1 können nicht verlangt werden.“

**Comment by fgoe** [comment]
Der Nachfrage würde ich mich anschliessen. Ich hab das Szenario, dass 2 Module auf der Ostseite liegen und 2 Module auf der Westseite. Die Module Ostseite an einem HM800 und die Westseite auch an einem HM800.
Aktuell sind beide jeweils auf 300W max konfiguriert, wegen dem 600W Limit. 
Beim Wetter zur Zeit, wenn die Sonne mal da ist....und ich temporär mal das Limit wegmache, dann macht die Ostseite ca 450W und die Westseite dann ca 50W. Zu späterer Uhrzeit dreht sich das ganze dann um (recht steiles Dach).

Ich hatte auch schon den Gedanken des Threaderstellers, ob es möglich wäre der OpenDTU zu sagen "max aus allem WR sei xxx Watt, wie du das ausregelst ist dir überlassen, hole das Maximum raus".

**Reply by spcqike** [reply]
In deinen Fall hätte ich einfach je ein Ost und ein West Modul parallel an einen Eingang gehangen. Sofern sie die gleiche Spannung liefern natürlich. 

Hat es einen Grund, dass du zwei wechselrichter dafür nehmen willst?

**Reply by fgoe** [reply]
Uff...jetzt wo du es sagst. Manchmal sieht man die einfachste Lösung nicht. Irgendwie will man da im Kopf die "Seiten" zusammenhalten, aber rein vom elektrischen Aufbau her ist das ja vollkommen egal, ob ich die mische.
Die 2 Wechselrichter und die 4 Module waren ein fertiges Set.
Danke für den Anstoß.

**Reply by spcqike** [reply]
Ob es elektrisch immer egal ist, kann ich nicht genau sagen. Immerhin könnte der Kurzschlussstrom von 2 parallelen Modulen, auch in O-W Ausrichtung, manche Hoymiles-Modelle kurzzeitig überlasten. So liefern Module mit 54 Zellen bspw. knappe 14A, also insgesamt bis zu 28A (vielleicht zur Mittagszeit, wenn beide recht gut beschienen sind), die HM-600 bis HM-800 haben aber nur 11.5-12.5A Eingangsstrom. DIe HMS-1600 bis HMS-2000 haben 14-16A Eingangsstrom und auch einen benannten Kurzschlussstrom von 25A. 

da viele User die Wechselrichter aber an Batterien betreiben, die weit mehr als 25A liefern können, und da die hier beispielhaften 28A der 2 Module wahrscheinlich nur sehr selten tatsächlich erreicht werden können, sehe ich da aber kein Risiko.
